src/quantilealgs.jl

```diff
@@ -30,93 +30,97 @@ quantile_bisect(d::ContinuousUnivariateDistribution, p::Float64) =
 #   Distribution, with Application to the Inverse Gaussian Distribution
 #   http://www.statsci.org/smyth/pubs/qinvgaussPreprint.pdf
 
-function quantile_newton(d::ContinuousUnivariateDistribution, p::Float64, xs::Float64=mode(d), tol::Float64=1e-12)
-    if 0.0 < p < 1.0
+function quantile_newton(d::ContinuousUnivariateDistribution, p::Real, xs::Real=mode(d), tol::Real=1e-12)
+    T = promote_type(partype(d), typeof(p), typeof(xs), typeof(tol))
+    if 0 < p < 1
         while true
             x = xs + (p - cdf(d, xs)) / pdf(d, xs)
             abs(x-xs) >= max(abs(x),abs(xs))*tol || return x
             xs = x
```

Comment:
Might cause trouble to reuse `xs` here when it is allowed to be any `Real` number. E.g. an `Int` would probably make the type of `xs` change and therefore it will end up being boxed. Probably a good idea to check functino with `@code_warntype` for various input types.